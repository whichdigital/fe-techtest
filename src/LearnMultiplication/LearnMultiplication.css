.main-container {
  margin-top: 20px;
  max-width: 700px;
}

.main {
  height: 75vh;
  overflow-y: auto;
}

.block {
  text-align: center;
  padding: 35px 0px;
  background-color: #6c757d;
  color: #fff;
  border: 2px solid #fff;
  cursor: pointer;
}

.block:hover {
  background-color: #999;
}

.highlight {
  background-color: dodgerblue;
  font-weight: bold;
}
