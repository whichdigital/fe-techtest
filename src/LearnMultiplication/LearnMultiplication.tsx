import { useState } from "react";
import { Button, Col, Container, Row } from "react-bootstrap";
import "./LearnMultiplication.css";

// interface for each number block
export interface BlockofNumbersPropsI {
  value: number;
  isMultiple: boolean;
}

// Predefined set of numbers
const numbers: BlockofNumbersPropsI[] = [];
for (let i = 1; i <= 144; i++) {
  numbers.push({
    value: i,
    isMultiple: false,
  });
}

const LearnMultiplication = () => {
  const [blockofNumbers, setBlockofNumbers] =
    useState<BlockofNumbersPropsI[]>(numbers);

  const calculateMultiples = (selectedNumber: number) => {
    // copy of setofnumbers with new array so component can rerender
    let numbersCopy = [...blockofNumbers];

    // check if number is multiple of selected number then set the flag
    numbersCopy?.forEach((n) => {
      if (selectedNumber && n.value % selectedNumber === 0) {
        n.isMultiple = true;
      } else {
        n.isMultiple = false;
      }
    });
    setBlockofNumbers(numbersCopy);
  };

  return (
    <Container className="main-container">
      <Row>
        <Col md={10}>
          <h3>Select a number to find it's multiples</h3>
        </Col>
        <Col md={2}>
          <Button type="reset" onClick={() => calculateMultiples(0)}>
            Reset
          </Button>
        </Col>
      </Row>
      <Row className="main" lg={3} md={2} sm={1} xs={1}>
        {blockofNumbers &&
          blockofNumbers.map((number: BlockofNumbersPropsI) => {
            return (
              <Col
                className={`block ${number.isMultiple ? "highlight" : ""}`}
                onClick={() => calculateMultiples(number.value)}
                key={number.value}
              >
                <Button
                  variant={number.isMultiple ? "success" : "secondary"}
                  type="button"
                  className={number.isMultiple ? "btn-outline-light" : ""}
                >
                  {number.value}
                </Button>
              </Col>
            );
          })}
      </Row>
    </Container>
  );
};

export default LearnMultiplication;
