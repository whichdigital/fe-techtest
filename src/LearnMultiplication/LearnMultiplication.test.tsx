import renderer from "react-test-renderer";
import LearnMultiplication from "./LearnMultiplication";

it("renders correctly", () => {
  const tree = renderer.create(<LearnMultiplication />).toJSON();
  expect(tree).toMatchSnapshot();
});
