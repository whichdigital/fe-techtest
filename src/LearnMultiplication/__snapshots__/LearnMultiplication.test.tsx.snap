// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main-container container"
>
  <div
    className="row"
  >
    <div
      className="col-md-10"
    >
      <h3>
        Select a number to find it's multiples
      </h3>
    </div>
    <div
      className="col-md-2"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="reset"
      >
        Reset
      </button>
    </div>
  </div>
  <div
    className="main row row-cols-lg-3 row-cols-md-2 row-cols-sm-1 row-cols-1"
  >
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        10
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        11
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        12
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        13
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        14
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        15
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        16
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        17
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        18
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        19
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        20
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        21
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        22
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        23
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        24
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        25
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        26
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        27
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        28
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        29
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        30
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        31
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        32
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        33
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        34
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        35
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        36
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        37
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        38
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        39
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        40
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        41
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        42
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        43
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        44
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        45
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        46
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        47
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        48
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        49
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        50
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        51
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        52
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        53
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        54
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        55
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        56
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        57
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        58
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        59
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        60
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        61
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        62
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        63
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        64
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        65
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        66
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        67
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        68
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        69
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        70
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        71
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        72
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        73
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        74
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        75
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        76
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        77
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        78
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        79
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        80
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        81
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        82
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        83
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        84
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        85
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        86
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        87
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        88
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        89
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        90
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        91
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        92
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        93
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        94
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        95
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        96
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        97
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        98
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        99
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        100
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        101
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        102
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        103
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        104
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        105
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        106
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        107
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        108
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        109
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        110
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        111
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        112
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        113
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        114
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        115
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        116
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        117
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        118
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        119
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        120
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        121
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        122
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        123
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        124
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        125
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        126
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        127
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        128
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        129
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        130
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        131
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        132
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        133
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        134
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        135
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        136
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        137
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        138
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        139
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        140
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        141
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        142
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        143
      </button>
    </div>
    <div
      className="block  col"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        144
      </button>
    </div>
  </div>
</div>
`;
